### PR TITLE
Improve recommendation on ERC721 transferFrom vs safeTransferFrom

### LINKED ISSUE #3477: Provide more nuance about ERC721's transferFrom vs safeTransferFrom
We have to mention the risk of reentrancy introduced by `safeTransferFrom`, and give more nuance to whatever we encourage or discourage.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/109778c17c7020618ea4e035efb9f0f9b82d43ca/contracts/token/ERC721/IERC721.sol#L84

### PR DESCRIPTION
**Pull request by goncaloMagalhaes**
Provide more nuance about ERC721's transferFrom vs safeTransferFrom
Fixes #3477

### CODE REVIEW

File: contracts/token/ERC721/IERC721.sol [hunk 1]
```diff
@@ -81,7 +81,9 @@ interface IERC721 is IERC165 {
     /**
      * @dev Transfers `tokenId` token from `from` to `to`.
      *
-     * WARNING: Usage of this method is discouraged, use {safeTransferFrom} whenever possible.
+     * WARNING: Note that the caller is responsible to confirm that the recipient is capable of receiving ERC721
+     * or else they may be permanently lost. Usage of {safeTransferFrom} is encouraged, though the caller must understand
```
Comment (frangio) [inline contracts/token/ERC721/IERC721.sol:85]: Let's avoid "encouraged".

```suggestion
     * or else they may be permanently lost. Usage of {safeTransferFrom} prevents loss, though the caller must understand
```

File: contracts/token/ERC721/IERC721.sol [hunk 2]
```diff
@@ -81,7 +81,9 @@ interface IERC721 is IERC165 {
...
+     * this adds an external call to {transferFrom}, potentially creating a reentrancy vulnerability.
```
Comment (frangio) [inline contracts/token/ERC721/IERC721.sol:86]: Current wording seems to imply that there is a call to `transferFrom`.

```suggestion
     * this adds an external call which potentially creating a reentrancy vulnerability.
```

**Review by frangio — COMMENTED** [review]
Thank you @goncaloMagalhaes!

File: contracts/token/ERC721/IERC721.sol (hunk 2 above)
Comment (goncaloMagalhaes) [reply contracts/token/ERC721/IERC721.sol:86]: You're right! I'll commit the changes addressing your feedback :)

**Review by frangio — DISMISSED** [review]
Thank you!

**Review by frangio — APPROVED** [review]

**Pull request merged**